src/Logary.Prometheus/Exporter.fs

```diff
@@ -0,0 +1,171 @@
+  let escapeHelp (help: string) =
+    help.Replace("\\", @"\\").Replace("\n",@"\n")
+
+  /// label_value can be any sequence of UTF-8 characters,
+  /// but the backslash (\), double-quote ("), and line feed (\n) characters
+  /// have to be escaped as \\, \", and \n, respectively.
+  let escapeLabelValue (labelValue: string) =
+    labelValue.Replace("\\", @"\\").Replace("\"",@"\""").Replace("\n",@"\n")
+
+  let formatFloat (value: float) =
+    if Double.IsPositiveInfinity value then "+Inf"
+    elif Double.IsNegativeInfinity value then "-Inf"
+    elif Double.IsNaN value then "Nan"
+    else value.ToString(CultureInfo.InvariantCulture)
+
+  let appendLine (sb: StringBuilder) s =
+    sb.AppendLine s |> ignore
+
+  let getTypeStr (metricInfo: MetricInfo) =
+    match metricInfo with
+    | Gauge _ -> "gauge"
+    | Histogram _ -> "histogram"
+
+  let getIdentifier nameTransform (name: string) (labels: Map<string,string>) =
+    let labelsStr =
+      if labels.Count = 0 then ""
+      else
+        let eachPairs =
+          labels
+          |> Seq.map (fun (KeyValue(name, value)) ->
+            sprintf "%s=\"%s\"" (nameTransform name) (escapeLabelValue value))
+        sprintf "{%s}" (String.Join(",", eachPairs))
+
+    sprintf "%s%s" name labelsStr
+
+  //  # HELP http_requests_total The total number of HTTP requests.
+  //  # TYPE http_requests_total counter
+  //  http_requests_total{method="post",code="200"} 1027 1395066363000
+  //  http_requests_total{method="post",code="400"}    3 1395066363000
+  //
+  //  # Escaping in label values:
+  //  msdos_file_access_time_seconds{path="C:\\DIR\\FILE.TXT",error="Cannot find file:\n\"FILE.TXT\""} 1.458255915e9
+  //
+  //  # Minimalistic line:
+  //  metric_without_timestamp_and_labels 12.47
+  //
+  //  # A weird metric from before the epoch:
+  //  something_weird{problem="division by zero"} +Inf -3982045
+  //
+  //  # A histogram, which has a pretty complex representation in the text format:
+  //  # HELP http_request_duration_seconds A histogram of the request duration.
+  //  # TYPE http_request_duration_seconds histogram
+  //  http_request_duration_seconds_bucket{le="0.05"} 24054
+  //  http_request_duration_seconds_bucket{le="0.1"} 33444
+  //  http_request_duration_seconds_bucket{le="0.2"} 100392
+  //  http_request_duration_seconds_bucket{le="0.5"} 129389
+  //  http_request_duration_seconds_bucket{le="1"} 133988
+  //  http_request_duration_seconds_bucket{le="+Inf"} 144320
+  //  http_request_duration_seconds_sum 53423
+  //  http_request_duration_seconds_count 144320
+  let exportMetricInfo (sb: StringBuilder) nameTransform (metricInfo: BasicInfo * seq<MetricInfo>) =
+    let basicInfo, metricInfos = metricInfo
+    let metricName = nameTransform basicInfo.name
+    let help = escapeLabelValue basicInfo.description
+    let metricType = metricInfos |> Seq.first |> Option.map getTypeStr
```

Comment:
This is double iteration of the seq. if this API is primarily internal for implementing functionality, consider using a concrete data-type that has O(1) indexing functionality, like an array.

Reply:
see explains below